### ISSUE TITLE
Discrete feature to create multiple AAD Group Owners

### ISSUE THREAD

**Issue by nitmatgeo** [hashicorp/terraform-provider-azuread#435]
Hi Team,
I would like to recommend to add another Resource: azuread_group_owners similar to [azuread_group_member](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/group_member). This will help to configure multiple owners or add/remove owners, otherwise, this is causing an issue when we use CSV files to create groups/owners.

Please let me know if this could be implemented, it should be easy and very similar to azuread_group_member.

**Comment by manicminer** [comment]
Hi @nitmatgeo, thanks for requesting. I've gone ahead and updated your initial comment using our issue template for feature requests as this aids us in reviewing.

You are correct this should be a relatively straightforward resource to implement. At the moment, the AzureAD provider is in a feature freeze whilst we implement major changes for the [ongoing Microsoft Graph transition](https://github.com/hashicorp/terraform-provider-azuread/issues/323). However, once we are able to merge new features we'll be able to work on this.

**Comment by manicminer** [comment]
Due to ongoing API issues I'm going to mark this one as blocked for now.

For context, since moving to MS Graph and then again after adding support for unified groups (aka M365/O365 groups) we've had to take great care to conform to new constraints on group ownership whilst maintaining compatibility for Terraform users who may or may not have permissions to read/write all groups in their tenant. At this time we are complying with these constraints, however reports continue of potential API errors, and so to avoid breaking existing configurations this feature will be on hold until we have further clarification. Thanks!

**Comment by pcornelissen** [comment]
Well, the problem is still present in recent azuread terraform + public azure-api combinations.
Right now I can't create a group via terraform, because I the duplicate owner error with http status 400.
If there is right now a problem that somehow the owner get's assigned twice somewhere along the call chain, why is that not silently discarded at least until the internal problem is fixed.
The intent is to create the group and the desired owner is clearly present in the request (although it's duplicated, but still the intent is crystal clear). Why can't you just ignore the duplicate value in this case and treat it as "set" instead of a list?

**Comment by sumitkatre123** [comment]
any news/update it is pushing us away from terraform and forcing to do things manually  ?

**Comment by manicminer** [comment]
This is blocked by various API issues including #1435
